plugins/woocommerce/client/admin/client/settings-payments/components/offline-gateway-list-item/offline-payment-gateway-list-item.tsx

```diff
@@ -28,35 +34,74 @@ type OfflinePaymentGatewayListItemProps = {
 
 export const OfflinePaymentGatewayListItem = ( {
 	gateway,
+	...props
 }: OfflinePaymentGatewayListItemProps ) => {
 	const [ isEnabled, setIsEnabled ] = useState( gateway.enabled );
 
-	return {
-		key: gateway.id,
-		title: <>{ gateway.title }</>,
-		className: 'transitions-disabled',
-		content: (
-			<span
-				dangerouslySetInnerHTML={ sanitizeHTML(
-					decodeEntities( gateway.content )
-				) }
-			/>
-		),
-		after: (
-			<PaymentGatewayButton
-				id={ gateway.id }
-				enabled={ isEnabled }
-				settings_url={ gateway.settings_url }
-				setIsEnabled={ setIsEnabled }
-			/>
-		),
-		before: (
-			<img
-				src={
-					gateway.square_image || gateway.image_72x72 || gateway.image
-				}
-				alt={ gateway.title + ' logo' }
-			/>
-		),
-	};
+	return (
+		<SortableItem
+			id={ gateway.id }
+			className="woocommerce-list__item woocommerce-list__item-enter-done"
+			{ ...props }
+		>
+			<div className="woocommerce-list__item-inner">
+				<div className="woocommerce-list__item-before">
+					<DefaultDragHandle />
+					<img
+						src={
+							gateway.square_image ||
+							gateway.image_72x72 ||
+							gateway.image
+						}
+						alt={ gateway.title + ' logo' }
+					/>
+				</div>
+				<div className="woocommerce-list__item-text">
+					<span className="woocommerce-list__item-title">
+						{ gateway.title }
+					</span>
+					<span
+						className="woocommerce-list__item-content"
+						dangerouslySetInnerHTML={ sanitizeHTML(
+							decodeEntities( gateway.content )
+						) }
+					/>
+				</div>
+				<div className="woocommerce-list__item-after">
+					<div className="woocommerce-list__item-after__actions">
+						<PaymentGatewayButton
+							id={ gateway.id }
+							enabled={ isEnabled }
+							settings_url={ gateway.settings_url }
+							setIsEnabled={ setIsEnabled }
+							needs_setup={ gateway.needs_setup }
+						/>
+					</div>
+				</div>
+			</div>
+		</SortableItem>
+	);
+};
+
+export const OfflinePaymentGatewayList = ( {
+	gateways,
+	setGateways,
+}: {
+	gateways: OfflinePaymentGateway[];
+	setGateways: ( gateways: OfflinePaymentGateway[] ) => void;
+} ) => {
+	return (
+		<SortableContainer< OfflinePaymentGateway >
```

Comment:
I've never seen type generics used inline with JSX component tags, TIL!

Reply:
Yes it's great! It was suggested by cursor 😄